docs/developer-guide/udf.rst

```diff
@@ -103,6 +103,14 @@ The KSQL server will check the value being passed to each parameter and report a
 log for any null values being passed to a primitive type. The associated column in the output row
 will be ``null``.
 
+Generics in UDFS
+~~~~~~~~~~~~~~~~
+
+A UDF declaration can utilize generics if they match the following conditions:
+
+- Any generic in the return value of a method must appear in at least one of the method parameters
+- The generic must not adhere to any interface. For example, ``<T extends Number>`` is not valid).
+
```

Comment:
We don't support type coercion for generics, but do for specific types, right? e.g. if I have a function like `add(T a, T b)` and I try to call it with something that resolves to `INT, BIGINT`, that won't work. But if I did `add(Long a, Long b)` it would. Might want to call that out in the docs.